text/0000-deprecation-ember-string.md

```diff
@@ -0,0 +1,62 @@
+- Start Date: 2017-07-14
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+This PR proposes the deprecation of `Ember.String`. The following methods will be moved to an addon:
+
+- `camelize`
+- `capitalize`
+- `classify`
+- `dasherize`
+- `decamelize`
+- `fmt`
+- `loc`
+- `underscore`
+- `w`
+
+The remaining methods (`htmlSafe` and `isHTMLSafe`) will be moved to `@ember/component`.
+
+In both cases, the String prototype won't be extended anymore.
```

Comment:
> You can almost do this with a regular expression, but a codemod can definitely be whipped up.

@locks not so sure about this one. You can detect easily in a codemod a pattern like `'foo bar'.dasherize()`. But hard to reliable transform a variable where we don't know what type it is (string or not), like:

```js
function do(something) {
  return something.dasherize();
}
```

`something` could be an instance of some class, having a `dasherize` method completely unrelated to `Ember.String`.

So I think transforming your codebase could be a mainly manual step. Not saying that we shouldn't do this anyways! ;)